algorithms/kernel/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -71,289 +92,249 @@ Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::compute(const Numer
-    }
+    // Maximal number of partial selections to be merged at once
+    const uint32_t selectionNumberOfChunks = 16;
+    const uint32_t histogramSize           = 256;
 
-    const size_t dataBlockSize  = 4096 * 4;
-    const size_t probeBlockSize = (2048 * 4) / sizeof(algorithmFpType);
-    const size_t nParts         = 16;
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(uint32_t, maxDataBlockRows, maxQueryBlockRows);
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(uint32_t, maxQueryBlockRows, nK);
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(uint32_t, maxQueryBlockRows * nK, selectionNumberOfChunks);
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(uint32_t, maxQueryBlockRows, histogramSize);
 
-    auto dataSq = context.allocate(TypeIds::id<algorithmFpType>(), dataBlockSize, &st);
-    DAAL_CHECK_STATUS_VAR(st);
-    auto data_temp = context.allocate(TypeIds::id<algorithmFpType>(), dataBlockSize * nFeatures, &st);
-    DAAL_CHECK_STATUS_VAR(st);
-    auto probe_temp = context.allocate(TypeIds::id<algorithmFpType>(), probeBlockSize * nFeatures, &st);
-    DAAL_CHECK_STATUS_VAR(st);
-    auto distances = context.allocate(TypeIds::id<algorithmFpType>(), dataBlockSize * probeBlockSize, &st);
-    DAAL_CHECK_STATUS_VAR(st);
-    auto categories = context.allocate(TypeIds::id<int>(), probeBlockSize * dataBlockSize, &st);
-    DAAL_CHECK_STATUS_VAR(st);
-    auto partialDistances = context.allocate(TypeIds::id<algorithmFpType>(), probeBlockSize * nK * (nParts + 1), &st);
+    auto dataSumOfSquares = context.allocate(TypeIds::id<algorithmFpType>(), maxDataBlockRows, &st);
     DAAL_CHECK_STATUS_VAR(st);
-    auto partialCategories = context.allocate(TypeIds::id<int>(), probeBlockSize * nK * (nParts + 1), &st);
+    auto distances = context.allocate(TypeIds::id<algorithmFpType>(), maxDataBlockRows * maxQueryBlockRows, &st);
     DAAL_CHECK_STATUS_VAR(st);
-    auto sorted = context.allocate(TypeIds::id<int>(), probeBlockSize * nK, &st);
+    auto partialDistances = context.allocate(TypeIds::id<algorithmFpType>(), maxQueryBlockRows * nK * selectionNumberOfChunks, &st);
     DAAL_CHECK_STATUS_VAR(st);
-    auto buffer = context.allocate(TypeIds::id<int>(), probeBlockSize * nK, &st);
+    auto partialLabels = context.allocate(TypeIds::id<int>(), maxQueryBlockRows * nK * selectionNumberOfChunks, &st);
     DAAL_CHECK_STATUS_VAR(st);
-    auto radix_buffer = context.allocate(TypeIds::id<int>(), probeBlockSize * 256, &st);
+    auto sorted = context.allocate(TypeIds::id<int>(), maxQueryBlockRows * nK, &st);
     DAAL_CHECK_STATUS_VAR(st);
-    auto classes = context.allocate(TypeIds::id<int>(), probeBlockSize, &st);
-    DAAL_CHECK_STATUS_VAR(st);
-    auto rndSeq = context.allocate(TypeIds::id<algorithmFpType>(), dataBlockSize, &st);
-    DAAL_CHECK_STATUS_VAR(st);
-    auto rndIndices = context.allocate(TypeIds::id<size_t>(), dataBlockSize, &st);
+    auto radix_buffer = context.allocate(TypeIds::id<int>(), maxQueryBlockRows * histogramSize, &st);
```

Comment:
What is the purpose of this buffer? It has not been read anywhere in `compute` method.